programs/fundamentals/cp.scl

```diff
@@ -0,0 +1,116 @@
+bin(a_num)->(
+if (a_num<=0,return([0]));
+res=[];
+while(a_num,4000,(
+    //print(a_num);
+    r=a_num%2;
+    put(res, null, r, 'insert');
+    a_num=(a_num-r)/2;
+    
+);
+
+);
+res
+);
+// 0 … 127 | 0xxxxxxx 
+// 128 … 2047 | 110xxxxx 10xxxxxx 
+// 2048 … 65535 | 1110xxxx 10xxxxxx 10xxxxxx 
+// 65536 … 1114111 | 11110xxx 10xxxxxx 10xxxxxx 10xxxxxx 
+chr(a_num,outer(__rmap))->(
+    digs=bin(a_num);
+    if(a_num<=2047,(
+        if(a_num<=127,(
+            res=[0,0,0,0,0,0,0,0];p=[7,6,5,4,3,2,1]
+        ),(
+            res=[1,1,0,0,0,0,0,0,1,0,0,0,0,0,0,0];p=[15, 14, 13, 12, 11, 10, 7, 6, 5, 4, 3]
+        ))
+    ),(
+        if(a_num<=65535,(
+            res=[1,1,1,0,0,0,0,0,1,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0];p=[23, 22, 21, 20, 19, 18, 15, 14, 13, 12, 11, 10, 7, 6, 5, 4]
+        ),(
+            res=[1,1,1,1,0,0,0,0,1,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0];p=[31, 30, 29, 28, 27, 26, 23, 22, 21, 20, 19, 18, 15, 14, 13, 12, 11, 10, 7, 6, 5]
+        ))
+    ));
+    //print(str(res)+'\n'+str(digs));
+    for(digs,(
+        res:(p:_i)=_
+    ));
+    //print(str(res));
+    tl='';
+    while(length(res)%6,50000,put(res, null, [0,0], 'extend');tl+='=');
+    ret='';
+    loop(length(res)/6,(
+        ret+=__rmap:(slice(res,0,6));
+        res=slice(res,6)
+    ));
+    decode_b64(ret+tl)
+
+
+);
+
+//test
+
+__test()->(
+    [ord('a'),ord('A'),ord_string('我'),ord('🚣'),ord('ř'),ord_string('1️⃣')];
+);
+__test2()->(
+    [ord_string('😊'),ord_string('❤️'),ord('🤎'),ord_string('💚'),ord_string('🇧🇷'),ord_string('👩‍❤️‍💋‍👨')]
+);
+__test3()->(
+    [chr(ord('🤎'))=='🤎',chr(ord('🚣'))=='🚣',chr(ord('ř'))=='ř',chr(ord('A')),chr(ord('a')),chr(ord('我')),chr(ord('😊'))=='😊']
+)
```

Comment:
Can you comment all of these? That way when someone imports the library they won't get the test functions, and they are only visible as examples and so we can test that the app works by uncommenting them.

Reply:
there is not /**/ in carpet script. so i use `if 0` to do that